controllers/users_controller.js

```diff
@@ -0,0 +1,35 @@
+const User = require('../models/users_model');
+
+const getAllUsers = async (req, res) => {
+    try {
+      const users = await User.find();
+      res.json(users);
+    } catch (err) {
+        console.warn("Error fetching users:", err);
+        res.status(500).json({ error: "An error occurred while fetching the users." });
+      }
+  };
+  
+const registerNewUser = async (req, res) => {
+  try {
+    const {username, email, password} = req.body;
+    const userExists = await User.findOne({username});
+    if (userExists) {
+      return res.status(400).json({ error: "username already exists." })
+    }
+
+    const user = new User({
+      username: username,
```

Comment:
```suggestion
      username,
```
When it's the same name, no need to write it twice.
Change `email` and `password` as well.